src/plugins/regions.ts

```diff
@@ -279,6 +282,23 @@ class SingleRegion extends EventEmitter<RegionEvents> {
       this.end = options.end ?? (isMarker ? this.start : this.end)
       this.renderPosition()
     }
+    if (options.id !== undefined && this.start !== undefined && this.end !== undefined) {
+      this.id = options.id
+      const isMarker = this.start === this.end
+      if (Array.isArray(this.id)) {
+        this.element.setAttribute('part', `${isMarker ? 'marker' : 'region'} ${this.id.join(' ')}`)
+      } else {
+        this.element.setAttribute('part', `${isMarker ? 'marker' : 'region'} ${this.id}`)
+      }
+    }
+    if (options.content) {
+      if (typeof options.content === 'string') {
+        this.element.innerHTML = options.content
```

Comment:
Not sure about `innerHTML` tbh. On one hand, it's convenient to pass an HTML string, on the other hand it's a security hole as you can pass any script like this creating an XSS vulnerability.

I would stick to either passing an HTML element or a pure text string using `content.textContent = options.content`.

Reply:
Im just going to do string, good point on the security flaws! This is my first js/ts project, so I may do some stupid stuff (I normally do python)

Reply:
Please don't resolve my comments, I will do so when they are addressed.